src/linker/implementation.py

```diff
@@ -0,0 +1,60 @@
+import os
+from pathlib import Path
+from typing import List
+
+from linker.configuration import Config
+from linker.utilities.general_utils import load_yaml
+
+
+class Implementation:
+    def __init__(self, step_name: str, config: Config, allowable_implementations: List[str]):
+        self.step_name = step_name
+        self.config = config
+        self.allowable_implentations = allowable_implementations
```

Comment:
```suggestion
        self.allowable_implementations = allowable_implementations
```

Typo, and it feels weird to me to have "allowable_implementations" in the `Implementation` instance. This seems like we haven't encapsulated something correctly.

Reply:
Added to separate PR